src/backend/marsha/core/api.py

```diff
@@ -497,25 +498,39 @@ def update_live_state(
         if not validate_signature(request.headers.get("X-Marsha-Signature"), msg):
             return Response("Forbidden", status=403)
 
-        video = self.get_object()
-        video.live_state = serializer.validated_data["state"]
-        live_info = video.live_info
-        live_info.update(
-            {"cloudwatch": {"logGroupName": serializer.validated_data["logGroupName"]}}
-        )
-
-        if video.live_state == defaults.RUNNING:
-            live_info.update({"started_at": stamp})
-            video.live_info = live_info
-
-        if video.live_state == defaults.STOPPED:
-            live_info.update({"stopped_at": stamp})
-            video.upload_state = defaults.HARVESTING
-            video.live_info = live_info
-            create_mediapackage_harvest_job(video)
-            delete_aws_element_stack(video)
+        with transaction.atomic():
+            try:
+                video = Video.objects.select_for_update(nowait=True).get(pk=pk)
+            except Video.DoesNotExist:
+                return Response({"success": False}, status=404)
+            except DatabaseError:
```

Comment:
are you sure that this will not catch another case e.g database down or transport error?

Reply:
Yes it definitely catch other database error. In the documentation they talk only about this execption https://docs.djangoproject.com/en/3.1/ref/models/querysets/#select-for-update

Reply:
I read this article : https://hakibenita.medium.com/how-to-manage-concurrency-in-django-models-b240fed4ee2

In you case, I think the optimistic approach is more appropriate. Maybe adding a state choice "DELETING" and doing something like:

```python
update_success = Video.objects.filter(pk=pk, upload_state="HARVESTED").update(upload_state="DELETING")
if update_success != 1:
    ... return 200
... delete the AWS stack etc.
video.upload_state="DELETED"
video.save()
```

Reply:
Thanks for the idea, I changed with his method